pkg/termination/termination_test.go

```diff
@@ -68,124 +73,128 @@ var _ = Describe("Handler Suite", func() {
-		})
-	})
-
-	Context("when a machine exists for the node", func() {
-		var counter int32
-		var testMachine *machinev1.Machine
-
-		BeforeEach(func() {
-			testMachine = newTestMachine("test-machine", testNamespace, nodeName)
-			createMachine(testMachine)
-
-			// Ensure the polling logic is excercised in tests
-			httpHandler = newMockHTTPHandler(func(rw http.ResponseWriter, req *http.Request) {
-				if atomic.LoadInt32(&counter) == 4 {
-					rw.Write([]byte("TRUE"))
-				} else {
-					atomic.AddInt32(&counter, 1)
-					rw.Write([]byte("FALSE"))
-				}
-			})
-		})
-
+	Context("when running the handler", func() {
 		JustBeforeEach(func() {
-			// Ensure the polling logic is excercised in tests
-			for atomic.LoadInt32(&counter) < 4 {
-				continue
-			}
+			stop, errs = StartTestHandler(h)
 		})
 
-		Context("and the handler is stopped", func() {
+		Context("when the handler is stopped", func() {
 			JustBeforeEach(func() {
 				close(stop)
 			})
 
 			It("should not return an error", func() {
 				Eventually(errs).Should(Receive(BeNil()))
 			})
+		})
 
-			It("should not delete the machine", func() {
-				key := client.ObjectKey{Namespace: testMachine.Namespace, Name: testMachine.Name}
-				Consistently(func() error {
-					m := &machinev1.Machine{}
-					return k8sClient.Get(ctx, key, m)
-				}).Should(Succeed())
+		Context("when no machine exists for the node", func() {
+			It("should return an error upon starting", func() {
+				Eventually(errs).Should(Receive(MatchError("error fetching machine for node (\"test-node\"): machine not found for node \"test-node\"")))
 			})
 		})
 
-		Context("and the instance termination notice is fulfilled", func() {
-			It("should delete the machine", func() {
-				key := client.ObjectKey{Namespace: testMachine.Namespace, Name: testMachine.Name}
-				Eventually(func() error {
-					m := &machinev1.Machine{}
-					err := k8sClient.Get(ctx, key, m)
-					if err != nil && errors.IsNotFound(err) {
-						return nil
-					} else if err != nil {
-						return err
+		Context("when a machine exists for the node", func() {
+			var counter int32
+			var testMachine *machinev1.Machine
+
+			BeforeEach(func() {
+				testMachine = newTestMachine("test-machine", testNamespace, nodeName)
+				createMachine(testMachine)
+
+				// Ensure the polling logic is excercised in tests
+				httpHandler = newMockHTTPHandler(func(rw http.ResponseWriter, req *http.Request) {
+					if atomic.LoadInt32(&counter) == 4 {
+						rw.Write([]byte("TRUE"))
+					} else {
+						atomic.AddInt32(&counter, 1)
+						rw.Write([]byte("FALSE"))
 					}
-					return fmt.Errorf("machine not yet deleted")
-				}).Should(Succeed())
+				})
 			})
-		})
 
-		Context("and the instance termination notice is not fulfilled", func() {
-			BeforeEach(func() {
-				httpHandler = newMockHTTPHandler(notPreempted)
+			JustBeforeEach(func() {
+				// Ensure the polling logic is excercised in tests
+				for atomic.LoadInt32(&counter) < 4 {
```

Comment:
i just want to make sure i understand how this counter is working, is this ensuring that the mock termination service is accessed 4 times before the test will continue?

Reply:
Yeah that's just so know we've looped a few times before continuing. This change is actually just a whitespace change though, I suggest reviewing this PR with whitespace changes hidden as it's more obvious what I've done here

Reply:
thanks, i'll double check with whitespace